sql/wsrep_applier.cc

```diff
@@ -164,22 +166,17 @@ int wsrep_apply_events(THD*        thd,
     case FORMAT_DESCRIPTION_EVENT:
       wsrep_set_apply_format(thd, (Format_description_log_event*)ev);
       continue;
-#ifdef GTID_SUPPORT
-    case GTID_LOG_EVENT:
-    {
-      Gtid_log_event* gev= (Gtid_log_event*)ev;
-      if (gev->get_gno() == 0)
-      {
-        /* Skip GTID log event to make binlog to generate LTID on commit */
-        delete ev;
-        continue;
-      }
-    }
-#endif /* GTID_SUPPORT */
     default:
       break;
     }
 
+    if (!thd->variables.gtid_seq_no && (ev->get_type_code() == QUERY_EVENT))
+    { 
+      uint64 seqno= wsrep_gtid_server.seqno_inc();
```

Comment:
This `seqno_inc()` call may happen out of order because it is done outside of commit order critical section. Why it is needed?

Reply:
As discussed, we need this because of replicated DDL which don't increase local seqno.
We wan't to increase seqno only for QUERY event and if it is TOI.